Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -110,7 +111,7 @@ public void createPaymentAuthRequest(@NonNull final FragmentActivity activity,
                 callback.onVenmoPaymentAuthRequest(new VenmoPaymentAuthRequest.Failure(new BraintreeException(
                         "Cannot collect customer data when ECD is disabled. Enable this feature " +
                                 "in the Control Panel to collect this data.")));
-                braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-switch.failed");
+                braintreeClient.sendAnalyticsEvent(VenmoAnalytics.TOKENIZE_FAILED);
```

Comment:
On BT iOS (and PPCP Android & iOS) we moved to this "notify pattern" (see [BT iOS](https://github.com/braintree/braintree_ios/blob/441f6d903ec8fd027459d124e61abb0b1b99f230/Sources/BraintreeVenmo/BTVenmoClient.swift#L399-L421) & [PPCP Android ](https://github.com/paypal/paypal-android/blob/fe8d82a0174e4bf49aaa2af5b8acd91bf428d459/CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt#L183-L196)examples) for firing analytics. The goal is that each analytics string constant should only be referenced in 1 place in source code. 

The purpose was to reduce errors in our firing of the events, since we have previous SDK version with bugs for forgetting to send analytics in certain cases. This way we know that every time we return a callback result/cancel/error to the merchant, we're firing the right event (and only once).

What do you think of this?

Reply:
I think that's a good idea - except for Android since we have two different callbacks for the two different parts of the flow, we'd need 5 analytics helper methods (one success, and then a cancel and failure for each callback) - do you still think that would be helpful?

Reply:
On second thought, to make sure we always fire an analytics event with each completion callback, I think it makes sense even if it requires 5 methods